### DISCUSSION TITLE
Temporary Node

### DISCUSSION THREAD

**Discussion by adi90x** [k3s-io/k3s#3828] [Q&A]
Hello,

Not sure if it even exist un k8s but is there a way / solution to define a node as temporary ?
At the moment, after a node join it will stay in cluster until manually deleted, in fact question coule be is there a way to automatically remove some node when they are unavailable ? This would make even more sense for k3s as it is running on edge device that may join/quit the cluster more frequently 

Regards,

**Comment by brandond** [comment]
No, but you could definitely write a controller or even just a simple shell script to delete nodes that are NotReady for a particular period of time. This sort of thing would normally be handled by a cloud provider (deleting nodes from the cluster that do not exist in the cloud provider inventory).
